m2c/translate.py

```diff
@@ -4365,6 +4368,34 @@ def setup_planned_vars(
             stack_info.planned_vars[key] = var
 
 
+def get_declared_reg_types(
+    stack_info: StackInfo, regs: List[Register]
+) -> [Dict[Register, Type]]:
+    # Use a struct where the type of the member var_x represents the type of the register
+    reg_struct_name = f"_m2c_reg_{stack_info.function.name}"
+    reg_struct = stack_info.global_info.typepool.get_struct_by_tag_name(
+        reg_struct_name, stack_info.global_info.typemap
+    )
+    if reg_struct is None:
+        return {}
+    types = {}
+    for reg_field in reg_struct.fields:
+        corresponding_regs = [
+            r
+            for r in regs
+            if reg_field.name.endswith(str(r).removeprefix("$")) and not r.is_float()
```

Comment:
```py
reg_name = stack_info.function.reg_formatter.format(reg)
name == f"var_{reg_name}"
```
I think. Why the `and not r.is_float()`?

Actually, hm, instead of hard-coding variable name, it might be cleaner if this function just parsed out a struct field name -> type mapping, and then `add_register_var` did the register -> struct field name lookup, since that is the one who knows the variable name.